lib/SIL/SILFunctionBuilder.cpp

```diff
@@ -88,15 +88,23 @@ void SILFunctionBuilder::addFunctionAttributes(SILFunction *F,
         vjpName = SILDeclRef(vjpFn).mangle();
       // Get lowered argument indices.
       auto paramIndices = A->getParameterIndices();
-      if (!paramIndices)
-        continue;
+      assert(paramIndices && "Parameter indices should have been resolved");
       auto loweredParamIndices = paramIndices->getLowered(
           F->getASTContext(),
           decl->getInterfaceType()->castTo<AnyFunctionType>());
       SILAutoDiffIndices indices(/*source*/ 0, loweredParamIndices);
       auto silDiffAttr = SILDifferentiableAttr::create(
           M, indices, A->getRequirements(), M.allocateCopy(jvpName),
           M.allocateCopy(vjpName));
+#if NDEBUG
```

Comment:
```suggestion
#ifndef NDEBUG
```

Reply:
Good catch, thanks!

Reply:
Oops,`test/AutoDiff/differentiable_attr_silgen.swift` fails after this change. Investigating.